hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenMetadataSpecs.java

```diff
@@ -0,0 +1,265 @@
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.newKeyNamed;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withOpContext;
+import static com.hedera.services.bdd.spec.utilops.records.SnapshotMatchMode.NONDETERMINISTIC_TOKEN_NAMES;
+import static com.hedera.services.bdd.spec.utilops.records.SnapshotMatchMode.NONDETERMINISTIC_TRANSACTION_FEES;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INVALID_SIGNATURE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.METADATA_TOO_LONG;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+import static com.hederahashgraph.api.proto.java.TokenType.NON_FUNGIBLE_UNIQUE;
+
+import com.hedera.services.bdd.junit.HapiTest;
+import com.hedera.services.bdd.junit.HapiTestSuite;
+import com.hedera.services.bdd.spec.HapiSpec;
+import com.hedera.services.bdd.spec.transactions.TxnUtils;
+import com.hedera.services.bdd.suites.HapiSuite;
+import com.hederahashgraph.api.proto.java.TokenFreezeStatus;
+import com.hederahashgraph.api.proto.java.TokenKycStatus;
+import com.hederahashgraph.api.proto.java.TokenSupplyType;
+import com.hederahashgraph.api.proto.java.TokenType;
+import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Tag;
+
+/**
+ * Validates the {@code TokenCreate} and {@code TokenUpdate} transactions, specifically its:
+ * <ul>
+ *     <li>Metadata and MetadataKey values and behaviours.</li>
+ * </ul>
+ */
+@HapiTestSuite
+@Tag(TOKEN)
+public class TokenMetadataSpecs extends HapiSuite {
+    private static final Logger log = LogManager.getLogger(TokenMetadataSpecs.class);
+    private static final String PRIMARY = "primary";
+    private static final String NON_FUNGIBLE_UNIQUE_FINITE = "non-fungible-unique-finite";
+    private static final String AUTO_RENEW_ACCOUNT = "autoRenewAccount";
+    private static final String ADMIN_KEY = "adminKey";
+    private static final String SUPPLY_KEY = "supplyKey";
+    private static final String CREATE_TXN = "createTxn";
+    private static final String PAYER = "payer";
+    private static final String METADATA_KEY = "metadataKey";
+    private static final String PAUSE_KEY = "pauseKey";
+    private static final String FREEZE_KEY = "freezeKey";
+    private static final String KYC_KEY = "kycKey";
+    private static final String FEE_SCHEDULE_KEY = "feeScheduleKey";
+
+    private static String TOKEN_TREASURY = "treasury";
+
+    public static void main(String... args) {
+        new TokenMetadataSpecs().runSuiteSync();
+    }
+
+    @Override
+    public boolean canRunConcurrent() {
+        return true;
+    }
+
+    @Override
+    public List<HapiSpec> getSpecsInSuite() {
+        return List.of(
+                rejectsMetadataTooLong(),
+                creationRequiresAppropriateSigsHappyPath(),
+                creationRequiresAppropriateSigs(),
```

Comment:
This is more subjective and picky, but maybe name `creationRequiresAppropriateSigs()` to `creationDoesNotHaveRequiredSigs()`, in order to differentiate from `creationRequiresAppropriateSigsHappyPath()`? I had to check both methods to see what the difference was.

Reply:
Done! 👍